conductor-client/src/main/kotlin/com/openlattice/postgres/external/ExternalDatabasePermissioner.kt

```diff
@@ -0,0 +1,498 @@
+            it.aclKey[1]
+        }
+        val columnsById = organizationExternalDatabaseColumns.getAll(extTableColIds).values.associate {
+            AclKey( it.tableId, it.id) to TableColumn(it.organizationId, it.tableId, it.id)
+        }
+        updateExternalTablePermissions(action, externalTableColAcls, columnsById)
+    }
+
+    fun executePrivilegesUpdateOnPropertyTypes(action: Action, assemblyAcls: List<Acl> ){
+        val esids = assemblyAcls.aclKeysAsSet {
+            it.aclKey[0]
+        }
+        val aclKeyToTableCols = entitySets.submitToKeys(esids, GetOrganizationIdFromEntitySetEntryProcessor()).thenApplyAsync { esidToOrgId ->
+            assemblyAcls.associate {
+                it.aclKey to TableColumn(esidToOrgId.getValue(it.aclKey[0]), it.aclKey[0], it.aclKey[1])
+            }
+        }
+        updateAssemblyPermissions(action, assemblyAcls, aclKeyToTableCols.toCompletableFuture().get())
+    }
+
+    private fun mapAclsToExternalVsAssemblyColumnAcls(acls: List<Acl>): Map<SecurableObjectType, List<Acl>>{
+        val aclKeyIndex = acls.aclKeysAsSet { it.aclKey.index }.toTypedArray()
+
+        val allOrgExternalDBAclKeys: Set<AclKey> = securableObjectTypes.keySet(
+                getAclKeysOfObjectTypePredicate( aclKeyIndex, SecurableObjectType.OrganizationExternalDatabaseColumn)
+        )
+        val allOrgExternalDbAcls = acls.filter { allOrgExternalDBAclKeys.contains(it.aclKey) }
+
+        val allAssemblyAclKeys: Set<AclKey> = securableObjectTypes.keySet(
+                getAclKeysOfObjectTypePredicate( aclKeyIndex, SecurableObjectType.PropertyTypeInEntitySet)
+        )
+        val allAssemblyAcls = acls.filter { allAssemblyAclKeys.contains(it.aclKey) }
+
+        return mapOf(
+                SecurableObjectType.OrganizationExternalDatabaseColumn to allOrgExternalDbAcls,
+                SecurableObjectType.PropertyTypeInEntitySet to allAssemblyAcls
+        )
+    }
+
+    private fun getAclKeysOfObjectTypePredicate(
+            aclKeysIndexForm: Array<String>,
+            objectType: SecurableObjectType
+    ): Predicate<AclKey, SecurableObjectType> {
+        return Predicates.and<AclKey, SecurableObjectType>(
+                Predicates.`in`<Any, Any>(SecurableObjectTypeMapstore.ACL_KEY_INDEX, *aclKeysIndexForm),
+                Predicates.equal<Any, Any>(SecurableObjectTypeMapstore.SECURABLE_OBJECT_TYPE_INDEX, objectType )
+        )
+    }
+
+    private fun <T> Collection<Acl>.aclKeysAsSet( extraction: ( Acl ) -> T ): Set<T> {
+        return this.mapTo( HashSet(this.size) ) { extraction(it) }
+    }
+
+    override fun addPrincipalToPrincipals(sourcePrincipalAclKey: AclKey, targetPrincipalAclKeys: Set<AclKey>) {
+        if ( targetPrincipalAclKeys.isEmpty()) {
+            return
+        }
+        val grantTargets = dbCredentialService.getDbUsernames(targetPrincipalAclKeys)
+
+        val sourceRole = dbCredentialService.getDbUsername(sourcePrincipalAclKey)
```

Comment:
Yes. It would be better to rename chunks of DBCredentialService to represent that it is a mapping of role to credential and that the caller needs to know whether the role is an actual user/username to use it as such